_posts/2017-08-25-introducing-sandboxfs.md

```diff
@@ -3,50 +3,52 @@ layout: posts
 title: Introducing sandboxfs
 ---
 
-sandboxfs is a new sandboxing technique for Bazel. It's experimental and
-subject to change, but it's available now for you to check out!
+sandboxfs is a new project to improve the way sandboxing works in Bazel by
+making it more efficient and correct.  It's experimental and subject to
+change, but it's available now for you to check out! Read on for details.
 
 # Correct builds
 
 As our motto claims, *correctness* is an integral part of Bazel. To achieve
 correctness, builds must be *hermetic* and
 [*reproducible*](https://reproducible-builds.org/), which means that all
-actions invoked by Bazel should be run in a well-defined environment. In other
-words: we want actions to run within a sandbox.
+actions invoked by Bazel should be run in a well-defined environment. In
+other words: we want actions to run within a sandbox.
 
 Let's consider an example. Think of a `cc_library` target that specifies
-`foo.cc` as a source. The `cc_library` rule contains a
-*compilation action* that turns the `foo.cc` source file into a `foo.o` object
-file. This action will run clang, and it will need to:
+`foo.cc` as a source. This `cc_library` target contains a *compilation
+action* that turns the `foo.cc` source file into a `foo.o` object file.
+This action will run clang, and it will need to:
 *  read `foo.cc`,
-*  read the transitive closure of all headers required by `foo.cc`
+*  read the transitive closure of all headers required by `foo.cc`,
 *  write `foo.o`, and
 *  possibly write some temporary files under `/tmp/`.
 
-To be confident this build is correct, Bazel must ensure that the action has
-read-only access to the identified input files and write-only access
-to the output and temporary files we expect. Otherwise, Bazel cannot know if the
-compiler went astray and read random files from the system, making future builds
-inconsistent.
+To be confident this build is correct, Bazel must ensure that the action
+has read-only access to the identified input files and write-only access to
+the output and temporary files we expect. Otherwise, Bazel cannot know if
+the compiler went astray and read random files from the system, making
+future builds inconsistent. One way to achieve this is by running each
```

Comment:
"One way to achieve this *what* is by"
Perhaps "this restriction", or "this environment"?

Using "this" by itself is grammatically okay, but it's always less ambiguous to specify with a noun. (Sometimes it's okay to let this edit go - later I left "this is slow" as-is.)

Reply:
Changed to "these restrictions".